editor/src/clj/editor/model_scene.clj

```diff
@@ -112,7 +113,7 @@
       :indexbuffer-format-16
       (let [short-buffer (.asShortBuffer byte-buffer)]
         (dotimes [i num-indices]
-          (aset out-indices i (int (.get short-buffer i)))))
+          (aset out-indices i (int (num/ushort->long (.get short-buffer i))))))
```

Comment:
We were already unpacking shorts to ints here, but we were putting in negative indices for short indices above the signed range. This change performs some bitwise operations to convert it the unsigned short into a signed long. There is no operation to convert into a signed int directly.

Note that the signed / unsigned problem remains for meshes that use 32-bit integers. However, I'm not sure there is a good solution there because Java arrays are addressed using signed ints, not longs.